### DISCUSSION TITLE
Ensembles in raspberry Pi

### DISCUSSION THREAD

**Discussion by subinsfernandez** [ensemblesaw/ensembles-app#84] [Q&A]
How can we install ensembles in raspberry Pi

**Comment by SubhadeepJasu** [comment]
I believe elementary has a pipeline for raspberry pi. I'll check if the app center apps already work there.

**Reply by subinsfernandez** [reply]
Ok

**Reply by subinsfernandez** [reply]
can we make this app as embedded os,

**Reply by subinsfernandez** [reply]
i have a plan to make my own arranger keyboad, i have no idea about software. could you pls help me.

**Reply by SubhadeepJasu** [reply]
I would love to. However, I don't have a RPI. I can modify the UI to just have the main display while all the other buttons are mapped to physical buttons. All this I can't test, may be you can.

**Reply by subinsfernandez** [reply]
👍🏻

**Reply by SubhadeepJasu** [reply]
Would it be possible to connect in slack or discord, so that we could work on this? (it will take me some time to get the next release out as I am currently in huge pressure due to certain life events)

**Reply by subinsfernandez** [reply]
I am an electrical engineering, have no idea about programming.
Currently trying to learn programming using internet.

**Reply by SubhadeepJasu** [reply]
I can take care of the programming part when it comes to this app. As for how to detect a physical button press in RPI I have no clue. One possibility is that I make all inputs to the app MIDI based and you can simply generate MIDI signals from the RPI.

**Reply by subinsfernandez** [reply]
Is it possible to make an OS using Linux kernel, configure Linux kernel and make your app working as OS

**Comment by subinsfernandez** [comment]
One possible way is to customize raspberry Pi GPIO pins or make a midi controller using Arduino microcontrollers and send midi signals to RPI via USB ports

**Reply by subinsfernandez** [reply]
https://discord.gg/r3WUUup5
